idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,4563 @@
+                * 1.75
+                / particle_sphericity
+                / bed_voidage_mf**3
+                * b.Reynolds_mf**2
+                + 10
+                * 150
+                * (1 - bed_voidage_mf)
+                / particle_sphericity**2
+                / bed_voidage_mf**3
+                * b.Reynolds_mf
+                == 10
+                * b.particle_dia**3
+                * b.dens_mass_gas
+                * (b.dens_mass_sol - b.dens_mass_gas)
+                * const.acceleration_gravity
+                / b.visc_d**2
+            )
+
+        # constraint for pressure drop
+        @self.Constraint(doc="Ergun equation for pressure drop")
+        def pressure_drop_eq(b):
+            particle_sphericity = 1.0
+            return (
+                b.pressure_drop * 1e-5
+                == (
+                    (
+                        -150
+                        * b.visc_d
+                        * (1 - b.bed_voidage) ** 2
+                        * b.velocity_in
+                        / b.bed_voidage**3
+                        / b.particle_dia**2
+                        / particle_sphericity**2
+                        - 1.75
+                        * (1 - b.bed_voidage)
+                        * b.velocity_in**2
+                        * b.dens_mass_gas
+                        / b.bed_voidage**3
+                        / b.particle_dia
+                        / particle_sphericity
+                    )
+                    * b.bed_height
+                )
+                * 1e-5
+            )
+
+        # add inlet port for fixed bed TSA
+        self._add_inlet_port()
+
+        # build up variables and constraints for heating step
+        self._add_heating_step()
+
+        # build up variables and constraints for cooling step
+        self._add_cooling_step()
+
+        # build up variables and constraints for pressurization step
+        self._add_pressurization_step()
+
+        # build up variables and constraints for adsorption step
+        self._add_adsorption_step()
+
+        # discretization of time domain
+        self._apply_transformation()
+
+        # add performance equations
+        self._make_performance()
+
+        # add outlet ports for fixed bed TSA
+        self._add_outlet_port()
+
+        # add emissions equations
+        self._emissions()
+
+        # add compressor
+        if self.config.compressor:
+            self._add_compressor()
+
+        # add steam calculation
+        if self.config.steam_calculation is not None:
+            self._add_steam_calc()
+
+    def _add_general_parameters(self):
+        """
+        Method to add general required parameters to run fixed bed TSA model.
+        This include fluid and wall column properties, component set,
+        etc.
+
+        """
+        self.component_list = Set(
+            initialize=["N2", "CO2", "H2O", "O2"], doc="List of components"
+        )
+        self.component_list_subset = Set(
```

Comment:
It would help to have a more descriptive name for these subsets - what is the reason for the two subsets?

Reply:
`component_list_subset` is for components involved in the isotherm calculations while components in `component_list_subset_2` are all sent to one outlet steam without detailed calculations. I renamed to `isotherm_components` and `other_components`.

Reply:
For safely, it might be better to have only one of these subsets and for everything else do `component_list-subset`. This way, you can never have a case where there is a component that falls throguh the cracks.